### DISCUSSION TITLE
Custom server redirection and ISR

### DISCUSSION THREAD

**Discussion by jorge-cs** [vercel/next.js#18486] [Help]
Not sure if this is a bug or a limitation so starting a discussion here as I think this is likely a very common scenario.

## Code example

https://github.com/jorge-cs/custom-server-isr-example/tree/master

## Introduction
We're very excited to use Incremental Static Regeneration on our site. Unfortunately our pages have georestrictions so, in order for that to work, we need to detect a header present on the request which poses a challenge as you're not allowed to access the query string (for obvious reasons) on getStaticProps.

As a solution, however, we assumed we could simply take our page structure and code the country as a dynamic route:

/pages/[country]/[slug].js

This, coupled with getStaticProps and getStaticPaths simply works.  You visit /us/myslug and you get the us version statically generated, you visit /es/myslug and you get the incrementally statically generated for spain.

## The custom server

While the solution above works, it means it is easily hacked, by having the country as part of the querystring, if you want to see georestricted content all you have to do is change the querystring and you're done, so obviously a no go.

We figured it would be easy to overcome, just use a custom server which will read the appropriate header and redirect to the appropriate place and you're done. Unfortunately this is where we started having problems:

```javascript

const server = express()
const country = getCountryFromHeader(req)

server.get('/a', (req, res) => {
    return app.render(req, res, `/${country}/a`, query)
  })
```

That code is simply saying, when you get a request to "a", read the country from the header and redirect to /country/a, however, when trying to run this we'll invariably get:

> Unhandled Runtime Error
> Error: The provided `as` value (/a) is incompatible with the `href` value (/[country]/a). Read more: https://err.sh/vercel/next.js/incompatible-href-as

Any ideas on how to solve this?

**Comment by fbudassi** [comment]
How did you overcome the problem?
I have a similar problem, decided to implement something quite similar, and got the same unhandled runtime error :cry:

**Reply by fbudassi** [reply]
btw, changing the fallback value in the getStaticPaths, I made it work, but I'm in next.js 9.5.5 and the feature is supposed to be unstable:

`fallback: 'unstable_blocking'`
